### PR TITLE
[FIX] point_of_sale: apply pricelist in self-ordering process

### PR DESCRIPTION
**Pull request by pedrambiria**
Before this commit, the pricelist was not considered when displaying product prices in the self-ordering, leading to incorrect pricing information. This commit ensures that the pricelist is correctly applied.

opw-4190268

---
I confirm I have signed the CLA and read the PR guidelines at www.odoo.com/submit-pr

### CODE REVIEW

File: addons/pos_self_order/models/product_product.py [hunk 1]
```diff
@@ -52,12 +52,31 @@ def _load_pos_self_data_fields(self, config_id):
 
     def _load_pos_self_data(self, data):
         domain = self._load_pos_self_data_domain(data)
-        fields = self._load_pos_self_data_fields(data['pos.config']['data'][0]['id'])
+        config_id = data['pos.config']['data'][0]['id']
+        fields = self._load_pos_self_data_fields(config_id)
```
Comment (adgu-odoo) [inline addons/pos_self_order/models/product_product.py:56]: you do not have to change those two lines

Comment (adgu-odoo) [reply addons/pos_self_order/models/product_product.py:56]: or you can keep it and only pass the `config_id` to the `_compute_product_price_with_pricelist` which could be even better what do you think ?

**Review by adgu-odoo — CHANGES_REQUESTED** [review]
Very nice catch, can you just look at the comment and tell me what you think ? Thank you

File: addons/pos_self_order/models/product_product.py (hunk 1 above)
Comment (pedrambiria) [reply addons/pos_self_order/models/product_product.py:56]: @adgu-odoo Thanks for reviewing my code. It was my initial plan. Now, I've modified the code.

**Review by adgu-odoo — APPROVED** [review]

**Comment by adgu-odoo** [comment]
robodoo r+

**Pull request closed without merging**